### PR TITLE
fix to allow sample renaming to work correctly in GenomicsDBImport

### PR DESCRIPTION
**Pull request by lbergelson**
it seems like the only change necessary was in our code
removing pointless asserts
adding tests the deal with renaming

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java [hunk 1]
```diff
@@ -492,11 +492,11 @@ public Object onTraversalSuccess() {
                                                                                    final int batchSize, final int lowerSampleIndex) {
         final SortedMap<String, FeatureReader<VariantContext>> sampleToReaderMap = new TreeMap<>();
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java]: Add explicit docs to the sample name map argument clarifying whether the sample names must match the names in the header.

File: src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java [hunk 2]
```diff
@@ -261,12 +269,131 @@ private static File createInOrderSampleMap() {
...
+                .map( pair -> pair.getKey() + "\t" + pair.getValue())
+                .collect(Collectors.joining("\n")));
+    }
+
+    @DataProvider
+    public static Iterator<Object[]> getRenameCombinations() {
+        final Map<String,String> noRemapping = new LinkedHashMap<>();
+        noRemapping.put("s1", "s1");
+        noRemapping.put("s2", "s2");
+        noRemapping.put("s3", "s3");
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: We really could use a test case with more than a handful of samples -- say, 100. You could write a method to permute the values in a sample -> sample mapping.

File: src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java [hunk 3]
```diff
@@ -261,12 +269,131 @@ private static File createInOrderSampleMap() {
...
+        try(final GenomicsDBFeatureReader<VariantContext, PositionalBufferedStream> reader = getGenomicsDBFeatureReader(workspace, b37_reference_20_21)) {
+            final CloseableTribbleIterator<VariantContext> iterator = reader.iterator();
+            Assert.assertTrue(iterator.hasNext(), "expected to see a variant");
+            Assert.assertTrue(expectedSampleNames.size() > 0);
+            Assert.assertEquals(expectedSampleNames.size(), renamingMap.size());
+            iterator.forEachRemaining(vc -> {
+                   Assert.assertEquals(vc.getSampleNames(), expectedSampleNames);
+                expectedSampleNames.forEach( sample -> {
+                   Assert.assertEquals(vc.getGenotype(sample).getAnyAttribute(SAMPLE_NAME_KEY), renamingMap.get(sample));
+                   Assert.assertEquals(vc.getGenotype(sample).getAnyAttribute(ANOTHER_ATTRIBUTE_KEY), 10); //check another attribute just to make sure we're not mangling things
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: Have you confirmed that this test case fails without the fix to the tool?

File: src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java [hunk 4]
```diff
@@ -261,12 +269,131 @@ private static File createInOrderSampleMap() {
...
+        final Map<String,String> sameInput = new LinkedHashMap<>();
+        sameInput.put("s1", "s1");
+        sameInput.put("s2", "s1");
+        sameInput.put("s3", "s1");
+
+
+        final Map<String,String> sameInputWeirdOrder = new LinkedHashMap<>();
+        sameInputWeirdOrder.put("s3", "s1");
+        sameInputWeirdOrder.put("s1", "s1");
+        sameInputWeirdOrder.put("s2", "s1");
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: What currently happens if there are duplicate sample names in the sample name map (ie., on the left side of the mapping)? Add a separate test case for this with (I would hope) an expected exception.

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: We have a test for this in `GenomicsDBImportUnitTest.testBadInputFiles`

File: src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java (hunk 3 above)
Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: A subset of the test cases fail without the fix.  It only applies to 1 code path and it depends on the batching, so many pass with / without the fix.

File: src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java (hunk 1 above)
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java]: added

File: src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java (hunk 2 above)
Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java]: opened #3861

**Pull request merged**